pkg/collector/corechecks/cluster/orchestrator/collector_bundle.go

```diff
@@ -231,8 +250,14 @@ func (cb *CollectorBundle) Run(sender aggregator.Sender) {
 		runDuration := time.Since(runStartTime)
 		log.Debugf("Collector %s run stats: listed=%d processed=%d messages=%d duration=%s", collector.Metadata().FullName(), result.ResourcesListed, result.ResourcesProcessed, len(result.Result.MetadataMessages), runDuration)
 
-		orchestrator.SetCacheStats(result.ResourcesListed, len(result.Result.MetadataMessages), collector.Metadata().NodeType)
-		sender.OrchestratorMetadata(result.Result.MetadataMessages, cb.check.clusterID, int(collector.Metadata().NodeType))
+		nt := collector.Metadata().NodeType
+
+		orchestrator.SetCacheStats(result.ResourcesListed, len(result.Result.MetadataMessages), nt)
+
+		if !orchestrator.IsCRDType(nt) { // for CR and CRD we don't have metadata but only manifests
```

Comment:
Any opinions? Should we make the cleaner?

Reply:
I agree that the nil check might be more bug-prone in the future.

Reply:
Thanks for the reviews. I like that idea more instead of the isCRD check, implemented here: f79a111442